src/Mod/Path/PathScripts/PathPost.py

```diff
@@ -216,6 +218,170 @@ def exportObjectsWith(self, objs, job, needFilename=True):
+                    curlist.append((obj, tc))
+
+                if tc != currTool and currTool is None:  # first TC
+                    currTool = tc
+                    curlist.append((obj, tc))
+                    continue
+
+                if tc != currTool and currTool is not None:  # TC changed
+                    if tc.ToolNumber == currTool.ToolNumber:  # Same tool /diff params
+                        curlist.append((obj, tc))
+                        currTool = tc
+                    else:  # Actual Toolchange
+                        # dump current state to postlist
+                        sublist = []
+                        t = None
+                        for fixture in fixturelist:
+                            sublist.append(fixture)
+                            for item in curlist:
+                                if item[1] == t:
+                                    sublist.append(item[0])
+                                else:
+                                    sublist.append(item[1])
+                                    t = item[1]
+                                    sublist.append(item[0])
+
+                        postlist.append(sublist)
+
+                        # set up for next tool group
+                        currTool = tc
+                        curlist = [(obj, tc)]
+
+            # flush remaining curlist to output
+            sublist = []
+            t = None
+            for fixture in fixturelist:
+                sublist.append(fixture)
+                for item in curlist:
+                    if item[1] == t:
+                        sublist.append(item[0])
+                    else:
+                        sublist.append(item[1])
+                        t = item[1]
+                        sublist.append(item[0])
+            postlist.append(sublist)
+
+        elif orderby == "Operation":
+            PathLog.debug("Ordering by Operation")
+            # Order by operation means ops are done in each fixture in
+            # sequence.
+            currTool = None
+            # firstFixture = True
+
+            # Now generate the gcode
+            for obj in job.Operations.Group:
+                scratchpad = []
+                tc = PathUtil.toolControllerForOp(obj)
+                if not PathUtil.opProperty(obj, "Active"):
+                    continue
+
+                PathLog.debug("obj: {}".format(obj.Name))
+                for f in fixturelist:
+
+                    scratchpad.append((f, None))
+                    scratchpad.append((obj, tc))
+
+                sublist = []
+                temptool = None
+                for item in scratchpad:
+                    if item[1] in [temptool, None]:
+                        sublist.append(item[0])
+                    else:
+                        sublist.append(item[1])
+                        temptool = item[1]
+                        sublist.append(item[0])
+                postlist.append(sublist)
```

Comment:
I thought the same thing but didn't want to go any farther for 0.20.   This change is already going to cause a merge conflict with #6603 which I'll need to resolve.  Since I'm waiting on #6603 until 0.21, I'll wait on splitting it and do both at the same time.